doc/releases/index.rst

```diff
@@ -1,14 +1,52 @@
 .. _zephyr_release_notes:
 
-Release Notes
-#############
+Releases
+########
 
 Zephyr project is provided as source code and build scripts for different
 target architectures and configurations, and not as a binary image. Updated
 versions of the Zephyr project are released approximately every three-months.
 
 All Zephyr project source code is maintained in a `GitHub repository`_.
 
+Release Life Cycle and Maintenance
+**********************************
+
+Periodic Releases
+=================
+
+The Zephyr project provides periodic releases every 4 months leading to the
+long term support releases approx. every 2 years. Priodic and non-LTS releases
+are maintained with updates, bug fixes and security related updates for at
+least two cycles, meaning that the project supports the most recent two
+releases in addition to the most recent LTS.
+
+Long Term Support and Maintenance
+=================================
+
+A Zephyr :ref:`Long Term Support (LTS) <release_process_lts>` release is
+published every 2 years and is branched and maintained independently from the
+main tree for at least 2.5 years after it was released.
+
+Support and maintenance for an LTS release stops at least half a year
+after the following LTS release is published.
+
+Security Fixes
+==============
+
+Each security issue fixed within zephyr shall be backported to the
+following releases:
+
+- The current Long Term Stable (LTS) release.
+
+- The most recent two releases.
```

Comment:
As a release manager for v3.1, I want to say that I didn't get notifications about security updates that were needed to be backported or when the releases should be cut. I think some security team feedback was needed but not provided.

Reply:
>  I didn't get notifications about security updates 

So I think this language should be changed to "should be backported [...]" to make it clear that this is a goal and not a guarantee.

Reply:
@ceolin can you commit to giving the notifications to current release manager this time? If so I am OK. If not I think this is false information and we should not publish it.

Reply:
> As a release manager for v3.1, I want to say that I didn't get notifications about security updates that were needed to be backported or when the releases should be cut. I think some security team feedback was needed but not provided.

Are you saying here that we had security issues that were not addressed in the 3.1 release? Can you provide some details about what was missed? and what happened? 
In any case, if that is true, we need to fix this, but I would not change documentation, policy and committment of the project because someone failed to provide information for a release, whether the reason might be.

Reply:
@mbolivar-nordic long story short, we had it in place when we were using JIRA, after we migrated to github our scripts broke and there were some information that github was working  on an API for advisories that, unfortunately, was not released yet. We've brought it recently to discuss, but to avoid this sort of problem I'll be doing it manually and will ensure that the policy is followed.